app/controllers/underpayments/UnderpaymentDetailSummaryController.scala

```diff
@@ -93,32 +93,48 @@ class UnderpaymentDetailSummaryController @Inject() (
         if (addAnother) {
           Future.successful(Redirect(controllers.underpayments.routes.UnderpaymentTypeController.onLoad()))
         } else {
-          (request.isRepFlow, request.checkMode) match {
-            case (true, _) => redirectForRepFlow()
-            case (_, true) => Future.successful(Redirect(controllers.cya.routes.CheckYourAnswersController.onLoad()))
-            case _ =>
-              if (request.isOneEntry) {
-                Future.successful(Redirect(controllers.reasons.routes.BoxGuidanceController.onLoad()))
-              } else {
-                Future.successful(Redirect(controllers.docUpload.routes.BulkUploadFileController.onLoad()))
-              }
+          cleanupVatAccounting(request).flatMap { updatedRequest =>
+            if (request.isRepFlow) {
+              redirectForRepFlow(updatedRequest)
+            } else if (request.checkMode) {
+              Future.successful(Redirect(controllers.cya.routes.CheckYourAnswersController.onLoad()))
+            } else if (request.dutyType == Vat) {
+              Future.successful(Redirect(controllers.underpayments.routes.PostponedVatAccountingController.onLoad()))
+            } else if (request.isOneEntry) {
+              Future.successful(Redirect(controllers.reasons.routes.BoxGuidanceController.onLoad()))
+            } else {
+              Future.successful(Redirect(controllers.docUpload.routes.BulkUploadFileController.onLoad()))
+            }
           }
         }
       }
     )
   }
 
-  private[underpayments] def redirectForRepFlow()(implicit request: DataRequest[_]): Future[Result] = {
+  private[underpayments] def cleanupVatAccounting(request: DataRequest[_]): Future[DataRequest[_]] = {
+    if (request.dutyType != Vat) {
+      for {
+        updatedAnswers <- Future.fromTry(request.userAnswers.remove(PostponedVatAccountingPage))
+        _              <- sessionRepository.set(updatedAnswers)
+      } yield request.copy(userAnswers = updatedAnswers)
+    } else {
+      Future.successful(request)
+    }
+  }
+
+  private[underpayments] def redirectForRepFlow(request: DataRequest[_]): Future[Result] = {
     val newUnderpaymentType: SelectedDutyType = request.dutyType
     val oldUnderpaymentType                   = request.userAnswers.get(TempUnderpaymentTypePage)
     val splitThePayment                       = request.userAnswers.get(SplitPaymentPage)
     val dutyOrVatOnly                         = Seq(Duty, Vat)
 
     (oldUnderpaymentType, newUnderpaymentType, splitThePayment) match {
       case (Some(oldType), Both, _) if dutyOrVatOnly.contains(oldType) =>
-        removePaymentDataAndRedirect()
+        removePaymentDataAndRedirect(request)
       case (Some(Both), newType, Some(true)) if dutyOrVatOnly.contains(newType) =>
-        removePaymentDataAndRedirect()
+        removePaymentDataAndRedirect(request)
+      case (None, Vat, _) =>
```

Comment:
The bulk question needs to go above the VAT question because Bulk users are no longer supposed to be shown the PVA page.